### PR TITLE
[Optimize](Variant) optimize schema update performance

### PR DESCRIPTION
**Pull request by eldenmoon**
When update schema with high concurrency, updaing schemas cost is expensive.
1. update schema only when rows is not 0
2. copy_from is expensive, use copy constructor

### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Thearas** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/olap/rowset_builder.cpp [hunk 1]
```diff
@@ -414,19 +415,7 @@ void BaseRowsetBuilder::_build_current_tablet_schema(int64_t index_id,
                                                     indexes[i], ori_tablet_schema);
     }
     if (_tablet_schema->schema_version() > ori_tablet_schema.schema_version()) {
-        // After schema change, should include extracted column
-        // For example: a table has two columns, k and v
-        // After adding a column v2, the schema version increases, max_version_schema needs to be updated.
-        // _tablet_schema includes k, v, and v2
-        // if v is a variant, need to add the columns decomposed from the v to the _tablet_schema.
-        if (_tablet_schema->num_variant_columns() > 0) {
```
Comment (csun5285) [inline be/src/olap/rowset_builder.cpp:422]: may be need to revert it

Comment (eldenmoon) [reply be/src/olap/rowset_builder.cpp]: done

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**